op-service/sources/l2_client.go

```diff
@@ -172,8 +163,7 @@ func (s *L2Client) SystemConfigByL2Hash(ctx context.Context, hash common.Hash) (
 
 	envelope, err := s.PayloadByHash(ctx, hash)
 	if err != nil {
-		// w%: wrap to preserve ethereum.NotFound case
-		return eth.SystemConfig{}, fmt.Errorf("failed to determine block-hash of hash %v, could not get payload: %w", hash, err)
+		return eth.SystemConfig{}, fmt.Errorf("failed to determine block-hash of hash %v, could not get payload: %w", hash, eth.MaybeAsNotFoundErr(err))
```

Comment:
The underlying `s.PayloadByHash` here should be the one inspecting the error, and we shouldn't have to do `eth.MaybeAsNotFoundErr` here.

Same applies to other methods here that use inner common getters.

Reply:
going one level deeper and checking at `(header|block|payload)Call()`